### PR TITLE
add support for win32

### PR DESCRIPTION
**Pull request by zhuzilin**
Hi!
I have added support for win32. It could function well both on npm start and npm run build.
The main modifies are:
  1. Added condition for app.hide() and app.dock.setBadge since they are not api for windows.
  2. Updated xo to pass the npm test in windows. (https://github.com/xojs/eslint-config-xo/pull/42)
  3. Modified package.json for packaging.
Hope you could merge my request. Thank you!
Sincerely,
Zilin

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,5 @@
 npm-debug.log
 node_modules/
 build/
+dist/
```
Comment (timche) [inline .gitignore]: `dist/` is not being created in any of the scripts(?)

File: package.json [hunk 2]
```diff
@@ -11,8 +11,7 @@
     "prettier": "prettier --no-semi --single-quote --write",
     "prettify": "npm run prettier -- src/*.js",
     "start": "electron .",
-    "build:macos": "electron-packager . --out=build --overwrite --platform=darwin --arch=x64 --app-version=$npm_package_version --icon=assets/icon.icns  --protocol-name=\"Email Address URL\" --protocol=\"mailto\"",
-    "build": "npm run build:macos"
+    "build": "electron-packager . --out=build --overwrite --icon=assets/icon.icns --protocol-name=\"Email Address URL\" --protocol=\"mailto\""
```
Comment (timche) [inline package.json]: Can you please add `--all`, so it builds all platforms? When I run this command on macOS, it only builds for `darwin`.

File: .gitignore (hunk 1 above)
Comment (zhuzilin) [reply .gitignore]: Oh, the dist/ was for electron-builder... not electron-packager... I will delete it...

**Comment by zhuzilin** [comment]
Hi,
I have changed those two files. And I also test the built version on linux x64 through VM, it should work.
Thank you.

**Comment by zhuzilin** [comment]
Of course! Deleted now.

**Comment by timche** [comment]
Thanks, merged! Will release asap.

**Comment by zhuzilin** [comment]
Thanks!

**Comment by timche** [comment]
@allcontributors[bot] add @zhuzilin for code

**Comment by timche** [comment]
@allcontributors[bot] add @zhuzilin for code

**Pull request merged**